pkg/assets/apps.go

```diff
@@ -115,3 +117,23 @@ func ApplyDaemonSets(apps []string, render RenderFunc, params RenderParams, kube
 	ds.Client = appsClient(kubeconfigPath)
 	return applyApps(apps, ds, render, params)
 }
+
+func DeleteDaemonSets(apps []string, render RenderFunc, params RenderParams, kubeconfigPath string) error {
```

Comment:
I think this will not work when:

1) microshift is stopped ( shutdown ) 
2) node boot with different IP address
3) microshift boots


I recommend instead during boot, we try to find incorrect daemonset configs (wrong IPs), and delete those before creating the new (or simply updating the old if that's possible).

Reply:
@mangelajo  Currently we don't specify the node ip address in ovnk daemonset, it is auto-detected in ovnk golang code, so the idea here is to restart ovnk daemonset and let it discover the node ip change automatically.

Re the second step above, ovnk pods will restart and pick the new node ip.

Reply:
The K8S_NODE env var in ovnk daemonset is from pod spec.nodeName, which will be updated by kubelet after microshift restart.

Reply:
@dhellmann oh, good point! we actually can get rid of the logic in a smart way :-) 
I updated the daemonset to use K8S_NODE_IP as env var.

Reply:
I liked this approach much more thanks @dhellmann & @zshi-redhat good work